masonry/src/text/render_text.rs

```diff
@@ -60,6 +61,63 @@ pub fn render_text(
                         }
                     }),
                 );
+            // We want to draw the underline/strikethrough on top of the glyphs, so we draw it later
+            if let Some(underline) = &style.underline {
+                let underline_brush = &brushes[underline.brush.0];
+                let run_metrics = glyph_run.run().metrics();
+                let offset = match underline.offset {
+                    Some(offset) => offset,
+                    None => run_metrics.underline_offset,
+                };
+                let width = match underline.size {
+                    Some(size) => size,
+                    None => run_metrics.underline_size,
+                };
+                // The `offset` is the distance from the baseline to the *top* of the underline
+                // so we move the line down by half the width
+                // Remember that we are using a y-down coordinate system
+                let y = glyph_run.baseline() - offset + width / 2.;
+
+                let line = Line::new(
+                    (glyph_run.offset() as f64, y as f64),
+                    ((glyph_run.offset() + glyph_run.advance()) as f64, y as f64),
+                );
+                scene.stroke(
+                    &Stroke::new(width.into()),
+                    transform,
+                    underline_brush,
+                    None,
+                    &line,
+                );
+            }
+            if let Some(strikethrough) = &style.strikethrough {
+                let strikethrough_brush = &brushes[strikethrough.brush.0];
+                let run_metrics = glyph_run.run().metrics();
+                let offset = match strikethrough.offset {
+                    Some(offset) => offset,
+                    None => run_metrics.strikethrough_offset,
+                };
+                let width = match strikethrough.size {
+                    Some(size) => size,
+                    None => run_metrics.strikethrough_size,
+                };
+                // The `offset` is the distance from the baseline to the *top* of the strikethrough
+                // so we move the line down by half the width
+                // Remember that we are using a y-down coordinate system
+                let y = glyph_run.baseline() - offset + width / 2.;
```

Comment:
Hmm. This seems to be too high. 
![image](https://github.com/user-attachments/assets/10736899-a636-448b-84f3-91ca792812b8)

Reply:
What if we just remove the width adjustment?

Reply:
https://learn.microsoft.com/en-us/typography/opentype/spec/os2#ystrikeoutposition does agree that the offset should be the offset of the top.

This is really confusing

Reply:
Yeah, the spec defines it that way, but doesn't really account for custom widths. Clearly Firefox is making _some_ adjustment here because the top varies by width.